testing-mysql-server-5/src/main/java/com/facebook/presto/testing/mysql/EmbeddedMySql5.java

```diff
@@ -30,31 +30,58 @@ public EmbeddedMySql5(MySqlOptions mySqlOptions)
     @Override
     public List<String> getInitializationArguments()
     {
-        return ImmutableList.of(
-                "--no-defaults",
-                "--initialize-insecure",
-                "--skip-sync-frm",
-                "--innodb-flush-method=nosync",
-                "--datadir", getDataDirectory());
+        ImmutableList<String> iList = ImmutableList.<String>builder()
+                .add(
+                    "--no-defaults",
+                    "--skip-sync-frm",
+                    "--innodb-flush-method=nosync",
+                    "--datadir=" + getDataDirectory())
+                .build();
+
+        if (isMariadb) {
+            return ImmutableList.<String>builder().addAll(iList).add("--basedir=" + getBaseDirectory()).build();
+        }
+        else {
+            return ImmutableList.<String>builder().addAll(iList).add("--initialize-insecure").build();
+        }
     }
 
     @Override
     public List<String> getStartArguments()
     {
-        return ImmutableList.of(
-                "--no-defaults",
-                "--skip-ssl",
-                "--default-time-zone=+00:00",
-                "--disable-partition-engine-check",
-                "--explicit_defaults_for_timestamp",
-                "--skip-sync-frm",
-                "--innodb-flush-method=nosync",
-                "--innodb-flush-log-at-trx-commit=0",
-                "--innodb-doublewrite=0",
-                "--bind-address=localhost",
-                "--lc_messages_dir", getShareDirectory(),
-                "--socket", getSocketDirectory(),
-                "--port", String.valueOf(getPort()),
-                "--datadir", getDataDirectory());
+        ImmutableList<String> iList = ImmutableList.<String>builder()
+                .add(
+                    "--no-defaults",
+                    "--default-time-zone=+00:00",
+                    "--skip-sync-frm",
+                    "--innodb-flush-method=nosync",
+                    "--innodb-flush-log-at-trx-commit=0",
+                    "--innodb-doublewrite=0",
+                    "--bind-address=localhost",
+                    "--port=" + String.valueOf(getPort()),
+                    "--datadir=" + getDataDirectory(),
+                    "--socket=" + getSocketDirectory())
+                .build();
+
+        if (isMariadb) {
+            return ImmutableList.<String>builder()
```

Comment:
sorry, I meant use the ilist builder directly instead of building it and adding  it to another list builder.

Reply:
Done. :)